src/pages/Main/Carousel/Carousel.js

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from 'react';
+import * as S from './Carousel.style';
+
+const CAROUSEL_DATA = ['dohyeon', 'hwang', 'kyung', 'tony', 'lang'];
+const DUMMY_LENGTH = 2;
+
+const Carousel = () => {
+  const [imgIdx, setImgIdx] = useState(DUMMY_LENGTH);
+  const [isTransitionOn, setIsTransitionOn] = useState(true);
+  const [carouselInterval, setCarouselInterval] = useState();
+
+  useEffect(() => {
+    const interval = setInterval(handleNextBtn, 2000);
+    setCarouselInterval(interval);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (imgIdx === CAROUSEL_DATA.length + DUMMY_LENGTH) {
+      setTimeout(() => {
+        setImgIdx(DUMMY_LENGTH);
+        setIsTransitionOn(false);
+      }, 500);
+    } else if (imgIdx < DUMMY_LENGTH) {
+      setTimeout(() => {
+        setImgIdx(CAROUSEL_WITH_DUMMY.length - 1 - DUMMY_LENGTH);
+        setIsTransitionOn(false);
+      }, 500);
+    } else if (!isTransitionOn) {
+      setTimeout(() => {
+        setIsTransitionOn(true);
+      }, 10);
+    }
+  }, [imgIdx, isTransitionOn]);
+
+  const handleNextBtn = () => {
+    if (imgIdx === CAROUSEL_DATA.length + DUMMY_LENGTH) return;
+    setImgIdx((prev) => prev + 1);
+  };
+
+  const handlePrevBtn = () => {
+    if (imgIdx < DUMMY_LENGTH) return;
+    setImgIdx((prev) => prev - 1);
+  };
+
+  const CAROUSEL_WITH_DUMMY = [...CAROUSEL_DATA];
+
+  for (let i = 0; i < DUMMY_LENGTH; i++) {
+    CAROUSEL_WITH_DUMMY.push(CAROUSEL_DATA[i]);
+    CAROUSEL_WITH_DUMMY.unshift(CAROUSEL_DATA[CAROUSEL_DATA.length - 1 - i]);
+  }
```

Comment:
```suggestion
  const CAROUSEL_WITH_DUMMY = CAROUSEL_WITH_DUMMY.reduce((acc, cur, i) => {
    acc.push(cur);
    acc.unshift(cur[cur.length - 1 - i]);
  }, [...CAROUSEL_DATA])
```
직접 돌려보진 않았는데 대강 이런 느낌이면 되지 않을까요? ㅋㅋㅋ